pkg/build/graph/analysis/bc.go

```diff
@@ -64,6 +67,51 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.WarningSeverity,
+						Key:      MissingInputImageStreamWarning,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+					})
+
+					continue
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamTagNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					if _, ok := imageStream.Status.Tags[tagNode.ImageTag()]; !ok {
```

Comment:
> I disagree. it doesn't currently exist, you can't currently run that buildconfig. the fact that you happen to have something that, if run, would fix the situation, doesn't mean it's not broken, imo.

I would argue that what the user has created is structurally sound and should not produce a warning.  There is no additional action we'd expect the user to perform in order to have a valid build/image chain.  I think that warnings should lead to actions.

Step 2 of `oc status` is provide suggestions.  In the case where a bc does not point to the istag, the action would be "tag an image".  In the case where a bc does point to the istag, the action would be "do nothing".  Failures on the originating bc would be highlighted with failing builds.